core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -841,7 +841,7 @@ package object config {
   private[spark] val SHUFFLE_SORT_INIT_BUFFER_SIZE =
     ConfigBuilder("spark.shuffle.sort.initialBufferSize")
       .internal()
-      .intConf
+      .bytesConf(ByteUnit.BYTE)
       .checkValue(v => v > 0, "The value should be a positive integer.")
```

Comment:
We need to check if the input value exists in the integer range?

Reply:
Yeah,thanks